client/src/pages/FavoriteRestaurants.js

```diff
@@ -1,9 +1,59 @@
+import styled from 'styled-components';
+import RestaurantCard from '../components/RestaurantCard';
 
+export default function FavoriteRestaurants({
+    favoriteRestaurants,
+    updateFavorites,
+  }) {
+    const isFavorite = (restaurant) =>
+      favoriteRestaurants.some(
+        (favoriteRestaurant) => restaurant._id === favoriteRestaurant._id
+      )
 
-export default function FavoriteRestaurants(){
+    const removeFavoriteRestaurant = (restaurant) => {
+        if (isFavorite(restaurant)) {
+          updateFavorites(
+            favoriteRestaurants.filter(
+              (favoriteRestaurant) => favoriteRestaurant._id !== restaurant._id
+            )
+          )
+        }
+    }
+
+  
+
+  
     return (
-        <div>
-            <p>test</p>
-        </div>
-    )
-}
\ No newline at end of file
+      <Main>
+        <h1>Favorite Restaurants</h1>
+        <Wrapper>
+          {favoriteRestaurants.map((restaurant, index) => (
+            <RestaurantCard
+              key={restaurant._id + index}
+              restaurant={restaurant}
+              onAddToFavorites={() => removeFavoriteRestaurant(restaurant)}
+              isFavorite={() => isFavorite(restaurant)}
```

Comment:
In this case you can call  `isFavorite()` directly and pass along `true` or `false` – which is returned by the function – to the property.

```suggestion
              isFavorite={isFavorite(restaurant)}
```